src/shogun/CMakeLists.txt

```diff
@@ -63,15 +64,16 @@ FOREACH(template ${LIBSHOGUN_SRC_TMP})
 
     ADD_CUSTOM_COMMAND(OUTPUT ${generated_cpp}
         COMMAND ${CMAKE_COMMAND} -E copy_if_different ${template} ${generated_cpp}
-        COMMAND ${PYTHON_EXECUTABLE} ${generator_script} ${generated_cpp} -in ${CMAKE_CURRENT_BINARY_DIR}/headers_list.txt
+        COMMAND ${PYTHON_EXECUTABLE} ${generator_script} ${generated_cpp} ${generated_json}
```

Comment:
can you check the meta example include directory discovery mechanism we do using ctags